app/discovery/discovery.go

```diff
@@ -85,13 +87,12 @@ func (s *Service) Match(srv, src string) (string, bool) {
 
 	s.lock.RLock()
 	defer s.lock.RUnlock()
-	for _, m := range s.mappers {
-		if m.Server != "*" && m.Server != "" && m.Server != srv {
-			continue
-		}
-		dest := m.SrcMatch.ReplaceAllString(src, m.Dst)
-		if src != dest {
-			return dest, true
+	for _, srvName := range []string {"*", "", srv} {
```

Comment:
I think this slice should be flipped, i.e. starts with srv. Otherwise, the catch-all server may have priority other the exact server name match

Reply:
Makes sense, made "srv" first.

Reply:
Not sure what empty string is meant to represent, seems like a default value, so putting it as last over explicit "*". What do you think?